submissions/asaMitaka/FriendsApp/script.js

```diff
@@ -0,0 +1,237 @@
+    form.last.reset();
+    switch(event.target.value) {
+        case "ageMore":
+            sortedAgeMore();
+            break;
+        case "ageLess":
+            sortedAgeLess();
+            break;
+    }
+}
+
+function sortedAgeMore() {
+    friends.sort((a, b) => a.dob.age - b.dob.age);
+    renderAllItemsToPage(friends);
+}
+
+function sortedAgeLess() {
+    friends.sort((a, b) => b.dob.age - a.dob.age);
+    renderAllItemsToPage(friends);
+}
+
+function sortedAtoZLast() {
+    friends.sort((a, b) => a.name.last !== b.name.last ? a.name.last < b.name.last ? -1 : 1 : 0);
+    renderAllItemsToPage(friends);
+}
+
+function sortedZtoALast() {
+    friends.sort((a, b) => a.name.last !== b.name.last ? a.name.last > b.name.last ? -1 : 1 : 0);
+    renderAllItemsToPage(friends);
+}
+
+form.gender.addEventListener('click', function(e) {
+    if (e.target && e.target.classList.contains('gender')) {
+        checkedGender(e);
+    }
+});
+
+function checkedGender(event) {
+    switch (event.target.value) {
+        case 'male':
+            renderMale();
+            break;
+        case 'female':
+            renderFemale();
+            break;
+        case 'all':
+            renderAll();
+            break;
+    }
+}
+
+function renderMale() {
+    friends = friendsCopy.slice();
+    friends = friends.filter(item=> item.gender === 'male');
+    renderAllItemsToPage(friends);
+}
+
+function renderFemale() {
+    friends = friendsCopy.slice();
+    friends = friends.filter(item=> item.gender === 'female');
+    renderAllItemsToPage(friends);
+}
+
+function renderAll() {
+    friends = friendsCopy.slice();
+    renderAllItemsToPage(friends);
+}
+
+form.minMax.addEventListener('change', checkNumber);    
+form.minMax.addEventListener('change', findByAge);
+
+function checkNumber() {
+    if (minNumber.value < 10) {
+        minNumber.value = 10
+        return;
+    } 
+    
+    if (maxNumber.value > 100 && minNumber.value < 10) {
+        maxNumber.value = 100;
+        minNumber.value = 10;
+        return;
+    } 
+    
+    if (maxNumber.value > 100) {
+        maxNumber.value = 100;
+        return;
+    } 
+
+    findByAge();
+}
+
+function findByAge() {
+    friends = friendsCopy.slice();
+
+    if (form.age.ageLess.checked) {
+        sortedAgeLess();
+    }
+
+    if (form.age.ageMore.checked) {
+        sortedAgeMore();
+    }
+
+    if (form.last.lastAToZ.checked) {
+        sortedAtoZLast();
+    }
+
+    if (form.last.lastZToA.checked) {
+        sortedZtoALast();
+    }
+
+    friends = friends.filter(item => {
+        if (item.dob.age > minNumber.value && item.dob.age <= maxNumber.value) {
+            return item;
+        }
+    });
+
+    renderAllItemsToPage(friends);
+}
+
+form.search.search.addEventListener('input', searchFunc);
+
+function searchFunc() {
+    isSearch = true;
+    let val = form.search.search.value.toLowerCase();
+    if (val && val.length > 0) {
+        content.innerHTML = '';
```

Comment:
Why is this here? It's a searching function, it should return an array of users that match search input, not do things in the DOM.